### PR TITLE
[DO NOT MERGE] [numpy] Fix chapters 8, 9 and 13

### PR DESCRIPTION
**Pull request by reminisce**
Depend on the following PRs
https://github.com/apache/incubator-mxnet/pull/15237
https://github.com/apache/incubator-mxnet/pull/15246

### CODE REVIEW

File: d2l/model.py [hunk 1]
```diff
@@ -140,16 +140,16 @@ def forward(self, X, *args):
...
     if valid_length is None:
-        return X.softmax()
+        return npx.softmax(X)
     else:
         shape = X.shape
         if valid_length.ndim == 1:
             valid_length = valid_length.repeat(shape[1], axis=0)
         else:
             valid_length = valid_length.reshape((-1,))
-        X = nd.SequenceMask(X.reshape((-1, shape[-1])), valid_length, True,
```
Comment (mli) [inline d2l/model.py]: we'd better rename it to `sequence_mask` to match the naming convention.

File: d2l/train.py [hunk 2]
```diff
@@ -391,23 +391,24 @@ def translate_ch7(model, src_sentence, src_vocab, tgt_vocab, max_len, ctx):
...
+    enc_outputs = model.encoder(np.expand_dims(enc_X, axis=0), enc_valid_length)
     dec_state = model.decoder.init_state(enc_outputs, enc_valid_length)
-    dec_X = nd.array([tgt_vocab.bos], ctx=ctx).expand_dims(axis=0)
+    dec_X = np.expand_dims(np.array([tgt_vocab.bos], ctx=ctx), axis=0)
     predict_tokens = []
     for _ in range(max_len):
         Y, dec_state = model.decoder(dec_X, dec_state)
         dec_X = Y.argmax(axis=2)
-        py = dec_X.squeeze(axis=0).astype('int32').asscalar()
+        py = int(dec_X.squeeze(axis=0).astype('int32'))
```
Comment (mli) [inline d2l/train.py]: can we remove `astype('int32')`

File: d2l/model.py (hunk 1 above)
Comment (reminisce) [reply d2l/model.py]: Will do.

File: d2l/train.py (hunk 2 above)
Comment (reminisce) [reply d2l/train.py]: Yes, I missed it. Will remove it.

File: d2l/model.py (hunk 1 above)
Comment (reminisce) [reply d2l/model.py]: I will submit a PR dedicated for fixing all the nn op names after this is merged.

**Comment by reminisce** [comment]
Merged to numpy2.

**Pull request closed without merging**